### DISCUSSION TITLE
Should I build openssl and zlib too?

### DISCUSSION THREAD

**Discussion by meow464** [benfogle/crossenv#136] [Q&A / Support]
Is it ok to use the target and host provided openssl and zlib or is it known to cause problems?
Also is there an IRC or some other community?

**Comment by pmp-p** [comment]
Hi all
a bit of an answer for the first q
linking dynamically to host ssl is known to cause troubles at least on android and other Linux based os without proper use of library semantic versioning  in the library loader , eg android has no numbering at all ... you will have missing symbols anytime

I join for the second with @meow464 since I know of some communities built more around - packaging - cross compiled extensions like "Python-for-android" , "Beeware", "PyOdide" and plenty of personnal attempts including my own but i don't think there's any around the root cause of the problem (setuptools / pip / insufficient multiarch specification ) which crossenv try to wrap/fix ( i don't know it enough to say).
 Would it be usefull to start discussion ( ie start addressing and upstreaming real solutions ... ) if there isn't a joint effort already ?

**Comment by benfogle** [comment]
I wouldn't  expect it to cause problems (though as @pmp-p said, dynamic linking can be tricky on some systems). For something like Yocto or Buildroot-based distributions, using the target provided openssl or zlib would be a perfectly reasonable thing to do.

For the second question, I haven't set up any sort of IRC. I did raise some of the underlying issues on discuss.pythong.org, though I admittedly didn't follow through very well. (Thank you for reminding me to bring that up again.)

**Comment by pmp-p** [comment]
about IRC i've asked in #pypa on libera irc ( https://www.pypa.io/en/latest/ ). If it would be ok to invade with cross compiling / PEP517 draftings. A priori it would be OK since it's low traffic, though i did not mention it could involve unsupported platforms i think it would be a better place for experimentation than the official forums.
